Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -2224,23 +2224,23 @@ void FunctionDefinitionWriter::visit(AST::AbstractFloatLiteral& literal)
     NumberToStringBuffer buffer;
     WTF::numberToStringWithTrailingPoint(literal.value(), buffer);
 
-    m_stringBuilder.append(span(&buffer[0]));
+    m_stringBuilder.append(std::span { buffer });
```

Comment:
```suggestion
    m_stringBuilder.append(std::span { buffer }.first(strlen(buffer.data())));
```

Reply:
Moving the `std::span` construction into `WTF::numberToStringWithTrailingPoint()` was even better!